### PR TITLE
Allow the suite parser to skip adding resources

### PR DESCRIPTION
**Pull request by ctsims**
X-Requested (but can be pulled on its own) with CommCare Android

### CODE REVIEW

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 1]
```diff
@@ -75,19 +77,25 @@ public Suite parse() throws InvalidStructureException, IOException, XmlPullParse
...
                             Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
-                            table.addResource(r, table.getInstallers().getMediaInstaller(path), resourceGuid);
+                            if(!skipResources) {
+                                table.addResource(r, table.getInstallers().getMediaInstaller(path), resourceGuid);
+                            }
                         }
                     } else if (parser.getName().toLowerCase().equals("xform")) {
                         //skip xform stuff for now
                         parser.nextTag();
                         Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
```
Comment (wpride) [inline backend/src/org/commcare/xml/SuiteParser.java]: Can those line go inside the if() block?

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 2]
```diff
@@ -75,19 +77,25 @@ public Suite parse() throws InvalidStructureException, IOException, XmlPullParse
...
-                        table.addResource(r, table.getInstallers().getLocaleFileInstaller(localeKey), resourceGuid);
+                        if(!skipResources) {
+                            table.addResource(r, table.getInstallers().getLocaleFileInstaller(localeKey), resourceGuid);
+                        }
                     } else if (parser.getName().toLowerCase().equals("media")) {
                         String path = parser.getAttributeValue(null, "path");
                         //Can be an arbitrary number of resources inside of a media block.
                         while (this.nextTagInBlock("media")) {
                             Resource r = new ResourceParser(parser, maximumResourceAuthority).parse();
```
Comment (wpride) [inline backend/src/org/commcare/xml/SuiteParser.java]: and this line

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 3]
```diff
@@ -140,4 +148,12 @@ protected boolean inValidationMode() {
         return false;
     }
 
+    /**
+     * If set to true, the parser won't process adding incoming resources to the resource table.
+     * This is helpful if the suite is being processed during a non-install phase
+     * @param skipResources
```
Comment (phillipm) [inline backend/src/org/commcare/xml/SuiteParser.java]: Remove unused javadoc param entry

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 4]
```diff
@@ -33,6 +33,8 @@
     ResourceTable table;
     String resourceGuid;
 
+    private boolean skipResources = false;
```
Comment (phillipm) [inline backend/src/org/commcare/xml/SuiteParser.java]: Could fixture processing also be skipped on app init?

File: backend/src/org/commcare/xml/SuiteParser.java [hunk 5]
```diff
@@ -33,6 +33,8 @@
...
+
     public SuiteParser(InputStream suiteStream, ResourceTable table, String resourceGuid) throws IOException {
```
Comment (phillipm) [inline backend/src/org/commcare/xml/SuiteParser.java]: Will you add a class level doc? Something saying that this is used to used to build menus, entry actions, display details, get the app version, and install resources. I found myself wondering why we even run the suite parser on app init, and having such documentation would have answered it for me.

Comment (ctsims) [inline backend/src/org/commcare/xml/SuiteParser.java]: Sure, will add

File: backend/src/org/commcare/xml/SuiteParser.java (hunk 4 above)
Comment (ctsims) [inline backend/src/org/commcare/xml/SuiteParser.java]: Hm, plausible. The app fixtures are tiny right now, so I'm inclined to keep doing it until I have a chance to consider it more deeply

File: backend/src/org/commcare/xml/SuiteParser.java (hunk 1 above)
Comment (ctsims) [inline backend/src/org/commcare/xml/SuiteParser.java]: We're stream processing, so we can't in this case, we need to consume the full document in order to move on, and can't skip stuff.

**Comment by ctsims** [comment]
@phillipm Addressed your comments. @wpride Unfortunately we can't quite just skip due to the way we're streaming rather than processing the full DOM. We might be able to skip the elements, but I'm not sure it's actually much faster.

**Pull request merged**